core.coffee

```diff
@@ -36,6 +36,82 @@
 		'Options without $ prefixes (e.g. `options: filter: ...`) are deprecated, please use a $ prefix (eg `options: $filter: ...`).'
 	)
 
+	class Poll
+		subscribers:
+			error: []
+			data: []
+
+		stopped: false
+
+		intervalTime: 10000
+
+		constructor: (requestFn, intervalTime) ->
+			if intervalTime?
+				@intervalTime = intervalTime
+
+			@requestFn =  requestFn
+			@start()
+
+		setPollInterval: (intervalTime) ->
+			@intervalTime = intervalTime
+			@stop()
+			@start()
+
+		runRequest: ->
+			return if @stopped
+			@requestFn()
+			.then (response) =>
+				return if @stopped
+
+				# Catch errors in event subscribers so that they don't trigger
+				# the 'catch' below, and that subsequent subscribers will still
+				# be called
+				@subscribers.data.forEach (fn) ->
+					try fn(response)
+					catch error
+						console.error('Caught error in data event subscription:', error)
```

Comment:
Could you prefix these with `pinejs-client` or similar - it'll make it easier to know where it's coming from in larger projects

Reply:
sure thing